LetsDoStuff.WebApi/Controllers/AccountController.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using LetsDoStuff.Domain;
+using LetsDoStuff.Domain.Models;
+using LetsDoStuff.WebApi.SettingsForAuth;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+
+namespace LetsDoStuff.WebApi.Controllers
+{
+    [Route("api/account")]
+    public class AccountController : Controller
+    {
+        private LdsContext context;
+
+        public AccountController(LdsContext context)
+        {
+            this.context = context;
+        }
+
+        [HttpGet]
+        public string GetTestOutput()
+        {
+            return "test output";
+        }
+
+        [HttpPost("token")]
+        public IActionResult Token(string userLogin, string userPasseword)
+        {
+            var identity = GetIdentity(userLogin, userPasseword);
+
+            if (identity == null)
+            {
+                return BadRequest(new { errorText = "Invalid login or passeword" });
+            }
+
+            var now = DateTime.Now;
+
+            var jwt = new JwtSecurityToken(
+                issuer: AuthOptions.ISSUER,
+                audience: AuthOptions.AUDIENCE,
+                notBefore: now,
+                claims: identity.Claims,
+                expires: now.Add(TimeSpan.FromMinutes(AuthOptions.LIFETIME)),
+                signingCredentials: new SigningCredentials(AuthOptions.GetSymmetricSecurityKey(), SecurityAlgorithms.HmacSha256));
+            var encodedJwt = new JwtSecurityTokenHandler().WriteToken(jwt);
+
+            var response = new
+            {
+                access_token = encodedJwt,
+                login = identity.Name
+            };
+
+            return Json(response);
+        }
+
+        private ClaimsIdentity GetIdentity(string login, string passeword)
+        {
+            var user = context.Users.FirstOrDefault<User>(x => x.Login == login && x.Passeword == passeword);
+            if (user != null)
+            {
+                var claims = new List<Claim>
+                    {
+                        new Claim(ClaimsIdentity.DefaultNameClaimType, user.Login),
+                        new Claim(ClaimsIdentity.DefaultRoleClaimType, user.Role)
+                    };
+                ClaimsIdentity claimsIdentity =
+                    new ClaimsIdentity(claims, "Token", ClaimsIdentity.DefaultNameClaimType, ClaimsIdentity.DefaultRoleClaimType);
+                return claimsIdentity;
+            }
+
+            return null;
```

Comment:
Return the null is not a good idea in OOP world. Please, see the https://www.yegor256.com/2014/05/13/why-null-is-bad.html

Reply:
I've thought this before. I use the same logic here: this private method returns only claimsIdentity's type and the BadRequest is not allowed to be here. I return null here but I immediately chack this in the Token method.

Reply:
Sorry, that's my fault. You could return the `ActionResult<ClaimsIdentity>` instead of the `ClaimsIdentity` object.

More details
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/generic-type-parameters 
https://docs.microsoft.com/en-us/aspnet/core/security/authorization/claims?view=aspnetcore-3.1